src/extension/debug-page/DatabaseOps.tsx

```diff
@@ -0,0 +1,172 @@
+import { openDB, wrap } from 'idb'
+import React from 'react'
+import type { BackupFormat, Instance, ObjectStore } from './types'
+import typeson from './typeson'
+
+export const DatabaseOps: React.FC = () => {
+    const onBackup = async () => {
+        const payload = await backupAll()
+        if (payload === undefined) {
+            return
+        }
+        const timestamp = ((value: Date) => {
+            const values = [
+                value.getUTCFullYear(),
+                value.getUTCMonth() + 1,
+                value.getUTCDate(),
+                value.getUTCHours(),
+                value.getUTCMinutes(),
+                value.getUTCSeconds(),
+            ]
+            return values.map((value) => value.toString().padStart(2, '0')).join('')
+        })(new Date())
+        download(`maskbook-dump-${timestamp}.json`, payload)
+    }
+    const onRestore = async () => {
+        const file = await select()
+        if (file === undefined) {
+            return
+        }
+        const parsed: BackupFormat = await typeson.parse(await file.text())
+        await restoreAll(parsed)
+    }
+    const onClear = async () => {
+        const databases = await indexedDB.databases?.()
+        if (databases === undefined) {
+            return
+        }
+        await Promise.all(
+            databases.map(async ({ name }) => {
+                await timeout(wrap(indexedDB.deleteDatabase(name)), 500)
+                console.log(`clear ${name}`)
+            }),
+        )
+    }
+    return (
+        <section>
+            <p>
+                <button onClick={onBackup}>Backup Database</button>
+            </p>
+            <p>
+                <button onClick={onRestore}>Overwrite Database with backup</button>
+            </p>
+            <p>
+                <button onClick={onClear}>Clear Database</button>
+            </p>
+        </section>
+    )
+}
+
+function select() {
+    return new Promise<File | undefined>((resolve) => {
+        const element = document.createElement('input')
```

Comment:
I suggest using real input on the web page. This enables browser native drag-and-drop the file on the input controller.

Reply:
why?

Reply:
Friendly to use. You can have a try: an HTML `<input>` tag supports drag file into it.

Reply:
why need?

Reply:
Friendly to use